R/MzTabReader.R

```diff
@@ -360,17 +400,20 @@ getMSMSScans = function(identified_only = FALSE)
     res$mass.deviations..da. =  gsub(",", ";", res$mass.deviations..da., fixed = TRUE)
   }
   
- 
+  if (!"dp.modification" %in% colnames(res)) {
+    res$dp.modification = TRUE
+  }
+  
   #set reverse to needed values
   if ("reverse" %in% colnames(res)){
     res$reverse=(res$reverse=="decoy")
   }
   
   #set contaminant to TRUE/FALSE
-  if ("contaminant" %in% colnames(res)){
-    res$contaminant = (res$contaminant > 0)
+  if (!"contaminant" %in% colnames(res)){
+    res$contaminant = FALSE
   }
- 
+  res$contaminant = (res$contaminant > 0)
```

Comment:
handling of contaminants is done nicely in a block here. In the other method, the `if...` and the `res$contaminant = (res$contaminant > 0)` are far apart (same for decoy?) can you also make it a block there?